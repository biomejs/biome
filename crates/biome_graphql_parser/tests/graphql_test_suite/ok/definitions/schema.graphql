schema {
  query: MyQueryRootType
  mutation: MyMutationRootType
	subscription: MySubscriptionRootType
}

"decs" schema {
  query: MyQueryRootType
  mutation: MyMutationRootType
}

"""sth"""
schema {
  query: MyQueryRootType
}

schema {
  query: query
  mutation: mutation
	subscription: subscription
}
