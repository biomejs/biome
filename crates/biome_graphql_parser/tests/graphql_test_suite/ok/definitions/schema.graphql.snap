---
source: crates/biome_graphql_parser/tests/spec_test.rs
expression: snapshot
---
## Input
```graphql
schema {
  query: MyQueryRootType
  mutation: MyMutationRootType
	subscription: MySubscriptionRootType
}

"decs" schema {
  query: MyQueryRootType
  mutation: MyMutationRootType
}

"""sth"""
schema {
  query: MyQueryRootType
}

schema {
  query: query
  mutation: mutation
	subscription: subscription
}

```

## AST

```
GraphqlRoot {
    bom_token: missing (optional),
    definitions: GraphqlDefinitionList [
        GraphqlSchemaDefinition {
            description: missing (optional),
            schema_token: SCHEMA_KW@0..7 "schema" [] [Whitespace(" ")],
            directives: GraphqlDirectiveList [],
            l_curly_token: L_CURLY@7..8 "{" [] [],
            root_operation_type: GraphqlRootOperationTypeDefinitionList [
                GraphqlRootOperationTypeDefinition {
                    operation_type: GraphqlOperationType {
                        value_token: QUERY_KW@8..16 "query" [Newline("\n"), Whitespace("  ")] [],
                    },
                    colon_token: COLON@16..18 ":" [] [Whitespace(" ")],
                    named_type: GraphqlNamedType {
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@18..33 "MyQueryRootType" [] [],
                        },
                    },
                },
                GraphqlRootOperationTypeDefinition {
                    operation_type: GraphqlOperationType {
                        value_token: MUTATION_KW@33..44 "mutation" [Newline("\n"), Whitespace("  ")] [],
                    },
                    colon_token: COLON@44..46 ":" [] [Whitespace(" ")],
                    named_type: GraphqlNamedType {
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@46..64 "MyMutationRootType" [] [],
                        },
                    },
                },
                GraphqlRootOperationTypeDefinition {
                    operation_type: GraphqlOperationType {
                        value_token: SUBSCRIPTION_KW@64..78 "subscription" [Newline("\n"), Whitespace("\t")] [],
                    },
                    colon_token: COLON@78..80 ":" [] [Whitespace(" ")],
                    named_type: GraphqlNamedType {
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@80..102 "MySubscriptionRootType" [] [],
                        },
                    },
                },
            ],
            r_curly_token: R_CURLY@102..104 "}" [Newline("\n")] [],
        },
        GraphqlSchemaDefinition {
            description: GraphqlDescription {
                graphql_string_value: GraphqlStringValue {
                    graphql_string_literal_token: GRAPHQL_STRING_LITERAL@104..113 "\"decs\"" [Newline("\n"), Newline("\n")] [Whitespace(" ")],
                },
            },
            schema_token: SCHEMA_KW@113..120 "schema" [] [Whitespace(" ")],
            directives: GraphqlDirectiveList [],
            l_curly_token: L_CURLY@120..121 "{" [] [],
            root_operation_type: GraphqlRootOperationTypeDefinitionList [
                GraphqlRootOperationTypeDefinition {
                    operation_type: GraphqlOperationType {
                        value_token: QUERY_KW@121..129 "query" [Newline("\n"), Whitespace("  ")] [],
                    },
                    colon_token: COLON@129..131 ":" [] [Whitespace(" ")],
                    named_type: GraphqlNamedType {
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@131..146 "MyQueryRootType" [] [],
                        },
                    },
                },
                GraphqlRootOperationTypeDefinition {
                    operation_type: GraphqlOperationType {
                        value_token: MUTATION_KW@146..157 "mutation" [Newline("\n"), Whitespace("  ")] [],
                    },
                    colon_token: COLON@157..159 ":" [] [Whitespace(" ")],
                    named_type: GraphqlNamedType {
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@159..177 "MyMutationRootType" [] [],
                        },
                    },
                },
            ],
            r_curly_token: R_CURLY@177..179 "}" [Newline("\n")] [],
        },
        GraphqlSchemaDefinition {
            description: GraphqlDescription {
                graphql_string_value: GraphqlStringValue {
                    graphql_string_literal_token: GRAPHQL_STRING_LITERAL@179..190 "\"\"\"sth\"\"\"" [Newline("\n"), Newline("\n")] [],
                },
            },
            schema_token: SCHEMA_KW@190..198 "schema" [Newline("\n")] [Whitespace(" ")],
            directives: GraphqlDirectiveList [],
            l_curly_token: L_CURLY@198..199 "{" [] [],
            root_operation_type: GraphqlRootOperationTypeDefinitionList [
                GraphqlRootOperationTypeDefinition {
                    operation_type: GraphqlOperationType {
                        value_token: QUERY_KW@199..207 "query" [Newline("\n"), Whitespace("  ")] [],
                    },
                    colon_token: COLON@207..209 ":" [] [Whitespace(" ")],
                    named_type: GraphqlNamedType {
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@209..224 "MyQueryRootType" [] [],
                        },
                    },
                },
            ],
            r_curly_token: R_CURLY@224..226 "}" [Newline("\n")] [],
        },
        GraphqlSchemaDefinition {
            description: missing (optional),
            schema_token: SCHEMA_KW@226..235 "schema" [Newline("\n"), Newline("\n")] [Whitespace(" ")],
            directives: GraphqlDirectiveList [],
            l_curly_token: L_CURLY@235..236 "{" [] [],
            root_operation_type: GraphqlRootOperationTypeDefinitionList [
                GraphqlRootOperationTypeDefinition {
                    operation_type: GraphqlOperationType {
                        value_token: QUERY_KW@236..244 "query" [Newline("\n"), Whitespace("  ")] [],
                    },
                    colon_token: COLON@244..246 ":" [] [Whitespace(" ")],
                    named_type: GraphqlNamedType {
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@246..251 "query" [] [],
                        },
                    },
                },
                GraphqlRootOperationTypeDefinition {
                    operation_type: GraphqlOperationType {
                        value_token: MUTATION_KW@251..262 "mutation" [Newline("\n"), Whitespace("  ")] [],
                    },
                    colon_token: COLON@262..264 ":" [] [Whitespace(" ")],
                    named_type: GraphqlNamedType {
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@264..272 "mutation" [] [],
                        },
                    },
                },
                GraphqlRootOperationTypeDefinition {
                    operation_type: GraphqlOperationType {
                        value_token: SUBSCRIPTION_KW@272..286 "subscription" [Newline("\n"), Whitespace("\t")] [],
                    },
                    colon_token: COLON@286..288 ":" [] [Whitespace(" ")],
                    named_type: GraphqlNamedType {
                        name: GraphqlName {
                            value_token: GRAPHQL_NAME@288..300 "subscription" [] [],
                        },
                    },
                },
            ],
            r_curly_token: R_CURLY@300..302 "}" [Newline("\n")] [],
        },
    ],
    eof_token: EOF@302..303 "" [Newline("\n")] [],
}
```

## CST

```
0: GRAPHQL_ROOT@0..303
  0: (empty)
  1: GRAPHQL_DEFINITION_LIST@0..302
    0: GRAPHQL_SCHEMA_DEFINITION@0..104
      0: (empty)
      1: SCHEMA_KW@0..7 "schema" [] [Whitespace(" ")]
      2: GRAPHQL_DIRECTIVE_LIST@7..7
      3: L_CURLY@7..8 "{" [] []
      4: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION_LIST@8..102
        0: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION@8..33
          0: GRAPHQL_OPERATION_TYPE@8..16
            0: QUERY_KW@8..16 "query" [Newline("\n"), Whitespace("  ")] []
          1: COLON@16..18 ":" [] [Whitespace(" ")]
          2: GRAPHQL_NAMED_TYPE@18..33
            0: GRAPHQL_NAME@18..33
              0: GRAPHQL_NAME@18..33 "MyQueryRootType" [] []
        1: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION@33..64
          0: GRAPHQL_OPERATION_TYPE@33..44
            0: MUTATION_KW@33..44 "mutation" [Newline("\n"), Whitespace("  ")] []
          1: COLON@44..46 ":" [] [Whitespace(" ")]
          2: GRAPHQL_NAMED_TYPE@46..64
            0: GRAPHQL_NAME@46..64
              0: GRAPHQL_NAME@46..64 "MyMutationRootType" [] []
        2: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION@64..102
          0: GRAPHQL_OPERATION_TYPE@64..78
            0: SUBSCRIPTION_KW@64..78 "subscription" [Newline("\n"), Whitespace("\t")] []
          1: COLON@78..80 ":" [] [Whitespace(" ")]
          2: GRAPHQL_NAMED_TYPE@80..102
            0: GRAPHQL_NAME@80..102
              0: GRAPHQL_NAME@80..102 "MySubscriptionRootType" [] []
      5: R_CURLY@102..104 "}" [Newline("\n")] []
    1: GRAPHQL_SCHEMA_DEFINITION@104..179
      0: GRAPHQL_DESCRIPTION@104..113
        0: GRAPHQL_STRING_VALUE@104..113
          0: GRAPHQL_STRING_LITERAL@104..113 "\"decs\"" [Newline("\n"), Newline("\n")] [Whitespace(" ")]
      1: SCHEMA_KW@113..120 "schema" [] [Whitespace(" ")]
      2: GRAPHQL_DIRECTIVE_LIST@120..120
      3: L_CURLY@120..121 "{" [] []
      4: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION_LIST@121..177
        0: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION@121..146
          0: GRAPHQL_OPERATION_TYPE@121..129
            0: QUERY_KW@121..129 "query" [Newline("\n"), Whitespace("  ")] []
          1: COLON@129..131 ":" [] [Whitespace(" ")]
          2: GRAPHQL_NAMED_TYPE@131..146
            0: GRAPHQL_NAME@131..146
              0: GRAPHQL_NAME@131..146 "MyQueryRootType" [] []
        1: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION@146..177
          0: GRAPHQL_OPERATION_TYPE@146..157
            0: MUTATION_KW@146..157 "mutation" [Newline("\n"), Whitespace("  ")] []
          1: COLON@157..159 ":" [] [Whitespace(" ")]
          2: GRAPHQL_NAMED_TYPE@159..177
            0: GRAPHQL_NAME@159..177
              0: GRAPHQL_NAME@159..177 "MyMutationRootType" [] []
      5: R_CURLY@177..179 "}" [Newline("\n")] []
    2: GRAPHQL_SCHEMA_DEFINITION@179..226
      0: GRAPHQL_DESCRIPTION@179..190
        0: GRAPHQL_STRING_VALUE@179..190
          0: GRAPHQL_STRING_LITERAL@179..190 "\"\"\"sth\"\"\"" [Newline("\n"), Newline("\n")] []
      1: SCHEMA_KW@190..198 "schema" [Newline("\n")] [Whitespace(" ")]
      2: GRAPHQL_DIRECTIVE_LIST@198..198
      3: L_CURLY@198..199 "{" [] []
      4: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION_LIST@199..224
        0: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION@199..224
          0: GRAPHQL_OPERATION_TYPE@199..207
            0: QUERY_KW@199..207 "query" [Newline("\n"), Whitespace("  ")] []
          1: COLON@207..209 ":" [] [Whitespace(" ")]
          2: GRAPHQL_NAMED_TYPE@209..224
            0: GRAPHQL_NAME@209..224
              0: GRAPHQL_NAME@209..224 "MyQueryRootType" [] []
      5: R_CURLY@224..226 "}" [Newline("\n")] []
    3: GRAPHQL_SCHEMA_DEFINITION@226..302
      0: (empty)
      1: SCHEMA_KW@226..235 "schema" [Newline("\n"), Newline("\n")] [Whitespace(" ")]
      2: GRAPHQL_DIRECTIVE_LIST@235..235
      3: L_CURLY@235..236 "{" [] []
      4: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION_LIST@236..300
        0: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION@236..251
          0: GRAPHQL_OPERATION_TYPE@236..244
            0: QUERY_KW@236..244 "query" [Newline("\n"), Whitespace("  ")] []
          1: COLON@244..246 ":" [] [Whitespace(" ")]
          2: GRAPHQL_NAMED_TYPE@246..251
            0: GRAPHQL_NAME@246..251
              0: GRAPHQL_NAME@246..251 "query" [] []
        1: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION@251..272
          0: GRAPHQL_OPERATION_TYPE@251..262
            0: MUTATION_KW@251..262 "mutation" [Newline("\n"), Whitespace("  ")] []
          1: COLON@262..264 ":" [] [Whitespace(" ")]
          2: GRAPHQL_NAMED_TYPE@264..272
            0: GRAPHQL_NAME@264..272
              0: GRAPHQL_NAME@264..272 "mutation" [] []
        2: GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION@272..300
          0: GRAPHQL_OPERATION_TYPE@272..286
            0: SUBSCRIPTION_KW@272..286 "subscription" [Newline("\n"), Whitespace("\t")] []
          1: COLON@286..288 ":" [] [Whitespace(" ")]
          2: GRAPHQL_NAMED_TYPE@288..300
            0: GRAPHQL_NAME@288..300
              0: GRAPHQL_NAME@288..300 "subscription" [] []
      5: R_CURLY@300..302 "}" [Newline("\n")] []
  2: EOF@302..303 "" [Newline("\n")] []

```
